Intermittent compiler errors with CFFI/PyPy

`uwsgi_get_dot_h()` returns a string without a null-terminator when `UWSGI_ZLIB` is not defined: https://github.com/unbit/uwsgi/blob/2.0.10/core/uwsgi.c#L4743-L4769

When `UWSGI_ZLIB` is defined, it gets null-terminated:

```c
// add final null byte
uwsgi_buffer_append(ub, "\0", 1);
```

This causes intermittent compiler failures with PyPy, as the string returned will overflow into garbage:

```
__pycache__/_cffi__g155a00cfx914a51eb.c:4933:1: error: stray ‘\3’ in program

 ^
debug: OperationError:
debug:  operror-type: VerificationError
debug:  operror-value: CompileError: command 'cc' failed with exit status 1
```

--- a/core/uwsgi.c
+++ b/core/uwsgi.c
@@ -4849,7 +4849,7 @@ extern char *uwsgi_dot_h;
 char *uwsgi_get_dot_h() {
 	char *src = uwsgi_dot_h;
 	size_t len = strlen(src);
-        char *ptr = uwsgi_malloc(len / 2);
+        char *ptr = uwsgi_malloc((len / 2) + 1);
         char *base = ptr;
         size_t i;
         unsigned int u;
@@ -4870,6 +4870,9 @@ char *uwsgi_get_dot_h() {
 	base = ub->buf;
 	ub->buf = NULL;
 	uwsgi_buffer_destroy(ub);
+#else
+        // add final null byte
+        *ptr = '\0';
 #endif
 	return base;
 }
@@ -4882,7 +4885,7 @@ extern char *uwsgi_config_py;
 char *uwsgi_get_config_py() {
         char *src = uwsgi_config_py;
         size_t len = strlen(src);
-        char *ptr = uwsgi_malloc(len / 2);
+        char *ptr = uwsgi_malloc((len / 2) + 1);
         char *base = ptr;
         size_t i;
         unsigned int u;
@@ -4903,6 +4906,9 @@ char *uwsgi_get_config_py() {
         base = ub->buf;
         ub->buf = NULL;
         uwsgi_buffer_destroy(ub);
+#else
+        // add final null byte
+        *ptr = '\0';
 #endif
         return base;
 }

